src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -1265,7 +1260,7 @@ private function updateProductStatusByShopConstraint(int $productId, bool $isEna
             $command->setActive($isEnabled);
             $this->getCommandBus()->handle($command);
             $this->addFlash('success', $this->trans('The status has been successfully updated.', 'Admin.Notifications.Success'));
-        } catch (ProductException $e) {
+        } catch (Exception $e) {
```

Comment:
```suggestion
        } catch (ProductException $e) {
```

Reply:
can you explain why?
If we are not catching all exceptions, then we leave the possibility for errors to bubble up and break the page.
On the other hand, when we catch all of them, we have a fallback which at least shows any unmapped error in flash message "unexpected error bla bla", but doesn't break the page (which for me seems to be what we want, no?).
I remember some time ago this was required and even a dedicated PR was done to replace all the cought exceptions to generic ones and main argument was that even if we know what exceptions our core throws, we cannot know about module exceptions.
Btw this can easily throw ShopException like (ShopAssociationNotFoundException, which is not ProductException, but a sub of ShopException)